components/script/dom/dommatrix.rs

```diff
@@ -56,315 +57,331 @@ impl DOMMatrix {
     pub fn from_readonly(global: &GlobalScope, ro: &DOMMatrixReadOnly) -> DomRoot<Self> {
         Self::new(global, ro.is_2d(), ro.matrix().clone())
     }
+
+    // https://drafts.fxtf.org/geometry-1/#dom-dommatrix-fromfloat32array
+    #[allow(unsafe_code)]
+    pub fn FromFloat32Array(global: &GlobalScope, mut array: CustomAutoRooterGuard<Float32Array>) ->
+        Fallible<DomRoot<DOMMatrix>> {
+        let vec: Vec<f64> = unsafe { array.as_slice() }.iter().map(|&x| x as f64).collect();
+        DOMMatrix::Constructor_(global, vec)
+    }
+
+    // https://drafts.fxtf.org/geometry-1/#dom-dommatrix-fromfloat64array
+    #[allow(unsafe_code)]
+    pub fn FromFloat64Array(global: &GlobalScope, mut array: CustomAutoRooterGuard<Float64Array>) ->
+        Fallible<DomRoot<DOMMatrix>> {
+            let vec: Vec<f64> = unsafe { array.as_slice() }.to_vec();
```

Comment:
These can use the safe `array.to_vec()` APIs that were recently added.

Reply:
ohh ok ok that's good